src/platform/nrfconnect/DiagnosticDataProviderImplNrf.cpp

```diff
@@ -41,6 +41,24 @@ DiagnosticDataProviderImplNrf & DiagnosticDataProviderImplNrf::GetDefaultInstanc
     return sInstance;
 }
 
+
+// Disable all Watermarks methods since the feature map is set to 0
+CHIP_ERROR DiagnosticDataProviderImplNrf::ResetWatermarks()
```

Comment:
What does it mean that the feature map is set to 0? It is application-specific so a user might want to use this feature.

Reply:
So for our samples we don't want to support reset watermarks, since there always are issues with that (do we properly handle the reset watermarks and watermarks heap?). Hence we want to disable feature watermarks in Software diagnostics cluster. However due to implementation of software diagnostics server in matter sdk, we cannot just disable all watermarks command/attributes and set feature-map to 0 on zap level.
https://github.com/project-chip/connectedhomeip/issues/27099